### PR TITLE
Add H2ExtrasLeafIO and H2ExtrasInnerIO pages support for index.bin files

### PR DESCRIPTION
**Pull request by shishkovilja**

### CODE REVIEW

File: src/main/scala/ru/nizhikov/ignite/pagestat/PageStatistic.scala [hunk 1]
```diff
@@ -144,6 +145,36 @@ class PageStatistic(extLog: Boolean, checkIndexes: Boolean, pageSz: Int) {
 
                     stat(pageType)(1) += (maxCnt - cnt)*itemSz
                 }
+                else if (pageType >= T_H2_EX_REF_LEAF_START && pageType <= T_H2_EX_REF_LEAF_END) {
+                    // See BPlusIO#ITEMS_OFF
+                    val ITEMS_OFF = COMMON_HEADER_END + 2 + 8 + 8
+
+                    //See AbstractH2ExtrasLeafIO constructor and #getVersions and H2ExtrasLeafIO constructor
+                    val payload = pageType - T_H2_EX_REF_LEAF_START + 1
```
Comment (shishkovilja) [inline src/main/scala/ru/nizhikov/ignite/pagestat/PageStatistic.scala:153]: payload size difference

File: src/main/scala/ru/nizhikov/ignite/pagestat/PageStatistic.scala [hunk 2]
```diff
@@ -144,6 +145,36 @@ class PageStatistic(extLog: Boolean, checkIndexes: Boolean, pageSz: Int) {
...
+                else if (pageType >= T_H2_EX_REF_INNER_START && pageType <= T_H2_EX_REF_INNER_END) {
+                    // See BPlusIO#ITEMS_OFF
+                    val ITEMS_OFF = COMMON_HEADER_END + 2 + 8 + 8
+
+                    //See AbstractH2ExtrasLeafIO constructor and #getVersions and H2ExtrasLeafIO constructor
+                    val payload = pageType - T_H2_EX_REF_INNER_START + 1
+                    val itemSz = 8 + payload
+
+                    //See BPlusInnerIO#getMaxCount and BPlusIO#getCount
+                    val maxCnt = (pageSz - ITEMS_OFF - 8) / (itemSz + 8)
```
Comment (shishkovilja) [inline src/main/scala/ru/nizhikov/ignite/pagestat/PageStatistic.scala:172]: maxCnt difference

File: src/main/scala/ru/nizhikov/ignite/pagestat/PageStatistic.scala [hunk 3]
```diff
@@ -144,6 +145,36 @@ class PageStatistic(extLog: Boolean, checkIndexes: Boolean, pageSz: Int) {
...
+                    val cnt = page.takeShort(COMMON_HEADER_END)
+
+                    //See BPlusInnerIO#offset && #offset0
+                    stat(pageType)(1) += (maxCnt - cnt) * (itemSz + 8)
+                }
```
Comment (shishkovilja) [inline src/main/scala/ru/nizhikov/ignite/pagestat/PageStatistic.scala:177]: free size difference

**Comment by nizhikov** [comment]
fixed by e4ee30769978aab493c48a8f602da960cd8a1051

**Comment by nizhikov** [comment]
@shishkovilja Thanks for the helpfull contribution!

**Pull request closed without merging**